inference-engine/tests/functional/plugin/conformance/subgraphs_dumper/src/matchers/single_op.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2018-2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "matchers/single_op.hpp"
+#include "ngraph/ops.hpp"
+
+using namespace SubgraphsDumper;
+
+// TODO: Move to some utils?
+bool compare_constants_data(const std::shared_ptr<ngraph::op::Constant> &op,
+                            const std::shared_ptr<ngraph::op::Constant> &ref) {
+    switch (op->get_element_type()) {
+        case ngraph::element::Type_t::boolean:
+            if (op->cast_vector<bool>() != ref->cast_vector<bool>()) {
+                return false;
+            } else {
+                return true;
+            }
+        case ngraph::element::Type_t::bf16:
+            if (op->cast_vector<ngraph::bfloat16>() != ref->cast_vector<ngraph::bfloat16>()) {
```

Comment:
What with floating point comparing precision?

Reply:
Well, in current situation all of the constants which are taking part in comparison expected to contain integer-like data, but in general I agree that relying on `!=` operator may be risky. Need to compare data with some epsilon.

Reply:
Comparison based on std::numeric_limits::epsilon was implemented